src/Entity/FieldDefinitionProvider.php

```diff
@@ -44,6 +44,17 @@ public function getBaseFieldDefinition(array $civicrm_field) {
               $field = $this->getIntegerDefinition($civicrm_field);
             }
           }
+          elseif (isset($civicrm_field['data_type']) && $civicrm_field['data_type'] === 'ContactReference') {
+            $field = BaseFieldDefinition::create('entity_reference')
+              ->setSetting('target_type', 'civicrm_contact')
```

Comment:
Hi @puresyntax71 Thanks for the fix. I agree this is still in progress but just tried to apply it on our install and confirmed that it displays the name of the contact reference field correctly 👍 .Although, there seems to be a miss since anonymous users still see the id instead of name. I wonder if we should set the target_type in the above line to something else eg `civicrm_contact_reference` (for custom fields) and add it to `src/SupportedEntities.php` as well.

    $civicrm_entity_info['civicrm_contact_reference'] = $civicrm_entity_info['civicrm_contact'];
    $civicrm_entity_info['civicrm_contact_reference']['permissions']['view'] = ['view all contacts', 'access all custom data'];

Reply:
Created a PR to fix the above bug - https://github.com/eileenmcnaughton/civicrm_entity/pull/378 @puresyntax71 @jackrabbithanna